scripts/galera_new_cluster.sh

```diff
@@ -21,11 +21,11 @@ EOF
     exit 0
 fi
 
-systemctl set-environment _WSREP_NEW_CLUSTER='--wsrep-new-cluster' && \
+echo _WSREP_NEW_CLUSTER='--wsrep-new-cluster' > @mysqlunixdir@/"wsrep-new-cluster-${1:-mariadb}" && \
```

Comment:
`-` before `$` breaks it for mariadb@service

Maybe something like:
<pre>
SERVICE=${1:-mariadb@service}
FILE=@mysqlunixdir@/wsrep-new-cluster-${SERVICE}
echo _WSREP_NEW_CLUSTER='--wsrep-new-cluster' > "${FILE}" ...
..
systemctl start "$SERVICE"
rm -f "$FILE"
</pre>

And then have both mariadb.service.in and mariadb@service.in the same with:
<pre>
EnvironmentFile=-@mysqlunixdir@/wsrep-new-cluster-%n
EnvironmentFile=-@mysqlunixdir@/wsrep-start-position-%n
</pre>

Reply:
Let's have a look at what my code does. `${1:-mariadb}` evaluates to:
* `mariadb` when argument is omitted -> the main unit
* `mariadb` for `mariadb` -> the main unit
* `mariadb@foo` for `mariadb@foo` - > instantiated unit

Within the unit files I use `%N`, which is the full unit name with type suffix removed.

I still think this works as-is. Or did I get anything wrong?

Reply:
Ah, missed default expansion and a more specific indicator that the filename was incorrect. So mariadb.service.in needs `@mysqlunixdir@/wsrep-start-position-mariadb` rather than `@mysqlunixdir@/wsrep-start-position`.

Reply:
No, that's not true. The file `@mysqlunixdir@/wsrep-start-position` is generated in main unit in `StartExecPre=` without `-mariadb` suffix.

(In contrast to file `@mysqlunixdir@/wsrep-new-cluster-%N`, which has the suffix as it is generated in script `galera_new_cluster`.)

`mariadb.service`:
* `@mysqlunixdir@/wsrep-start-position`
* `@mysqlunixdir@/wsrep-new-cluster-%N`

`mariadb@.service`:
* `@mysqlunixdir@/wsrep-start-position-%N`
* `@mysqlunixdir@/wsrep-new-cluster-%N`

Reply:
oh, so sorry. Going to sleep now.